pyat/at/collective/wake_elements.py

```diff
@@ -67,6 +73,36 @@ def set_normfactxy(self, ring):
         l0, _, _ = ring.get_optics(ring)
         self.NormFact[0] = 1/l0['beta'][0]
         self.NormFact[1] = 1/l0['beta'][1]
+        
+    def get_opt_attr(self, attrname):
+        if hasattr(self, attrname):
+            return getattr(self, attrname)
+        else:
+            return None
+        
+    @property
+    def WakeT(self):
+        return self._wakeT
+        
+    @property
+    def WakeZ(self):
+        return self.get_opt_attr('_wakeZ')
+        
```

Comment:
```python
def WakeZ(self):
         return getattr(self, '_wakeZ', None)
```
The `get_opt_attr` method is inefficient!
Same comment for the similar properties.